src/validation-engine.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  ValidationResult,
+  FormValidationResult,
+  FieldsValidationSchema,
+  RecordValidationFunction,
+  createDefaultValidationResult,
+  RecordValidationFunctionSyncAsync,
+  isSyncValidationResult,
+  FieldValidationFunctionSyncAsync,
+  FieldValidationFunction,
+  FieldValidation,
+} from './model';
+
+import { isUndefinedOrNull } from './helper';
+
+import {
+  fireAllFieldsValidations,
+  fireSingleFieldValidations,
+  fireRecordValidations,
+} from './validation-dispatcher';
+import { buildFormValidationResult } from './form-validation-summary-builder';
+
+export class ValidationEngine {
+  private validationsPerField: FieldsValidationSchema = {};
+  private validationsGlobalForm: RecordValidationFunction[] = [];
+  private asyncValidationInProgressCount = 0;
+
+  // TODO, in new api we are adding an optional parameter
+  // to customize the error message (string array)
+  //
+  // TODO: Chain pattern applied here, check why it is useful
```

Comment:
I think we are not using it (in `lcFormValidation` I mean)

Reply:
Then we can remove it will do, if needed we can always add it again